validate.go

```diff
@@ -40,7 +94,13 @@ func validateAndAdjustContainerResourceRequests(container *corev1.Container, set
 
 func validateAndAdjustContainerResourceLimit(container *corev1.Container, resourceName string, resourceConfig *ResourceConfiguration) (bool, error) {
 	resourceStr, found := container.Resources.Limits[resourceName]
+	if resourceConfig.IgnoreValues && !found {
+		return false, fmt.Errorf("container does not have a %s limit", resourceName)
+	}
 	if !found || resourceStr == nil || len(strings.TrimSpace(string(*resourceStr))) == 0 {
+		if resourceConfig.IgnoreValues {
+			return false, fmt.Errorf("missing %s limit", resourceName)
+		}
```

Comment:
Now that you have other functions that is checking the resource fiel before calling this function, you do not need to check the `IgnoreValues` again. You can remove this if or remove the new functions.

Reply:
Nice catch... This WAS an issue until I moved around some unit tests. Thanks for catching this!